.github/workflows/ci.yml

```diff
@@ -99,6 +99,7 @@ jobs:
           codesign: '-'
           codesign-prefix: 'com.example.'
           codesign-options: 'runtime'
+          upx: ${{ startsWith(matrix.os, 'ubuntu-') && matrix.target != 'x86_64-pc-windows-gnu' && matrix.target != 'aarch64-unknown-linux-gnu' }}
```

Comment:
See https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/

> Is the reason it doesn't work with aarch64 and windows-gnu because UPX doesn't support cross-compilation?

See https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547228299?pr=90#step:6:296

> Is it intentional that this is only tested in linux? Have there been any problems when tested otherwise?

- [`win64/arm64` is not yet supported](https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547233777?pr=90#step:6:297)
- macOS is not supported

Reply:
> > Is the reason it doesn't work with aarch64 and windows-gnu because UPX doesn't support cross-compilation?
> 
> See https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547228299?pr=90#step:6:296

It seems to work with aarch64 linux-gnu + build-tool=cargo.
https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547228655?pr=90

> > Is it intentional that this is only tested in linux? Have there been any problems when tested otherwise?
> 
> * [`win64/arm64` is not yet supported](https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547233777?pr=90#step:6:297)
> * macOS is not supported

This doesn't seem to describe why x86_64 windows is skipped.
https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12377523218/job/34547233333?pr=90

Reply:
> It seems to work with aarch64 linux-gnu + build-tool=cargo.

Added.

> This doesn't seem to describe why x86_64 windows is skipped.

The output of the `file` command is the same when the exe is compressed with UPX or not:

```sh
./test-crate/target/release/test-crate.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
```

See [this run (compressing)](https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12516775448/job/34916413569#step:8:27) and [this run (not compressing)](https://github.com/taiki-e/upload-rust-binary-action/actions/runs/12516801417/job/34916496971#step:8:27). I don't know how to test it.